zulip_bots/zulip_bots/get_bot_output.py

```diff
@@ -0,0 +1,130 @@
+        spec = importlib.util.spec_from_file_location(name, path)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+        return module
+
+
+def parse_args():
+    usage = '''
+        get-bot-output <bot_name> --message "Send this message to the bot"
+        Example: get-bot-output xkcd --message 1
+        (This program loads bot-related code from the
+        library code and then feeds the message provided
+        in the command to the library code to handle.)
+        '''
+
+    parser = argparse.ArgumentParser(usage=usage)
+    parser.add_argument('name',
+                        action='store',
+                        nargs='?',
+                        default=None,
+                        help='the name of an existing bot to run')
+    parser.add_argument('--message',
+                        action='store',
+                        help='the message content to send to the bot')
+
+    parser.add_argument('--path-to-bot',
+                        action='store',
+                        help='path to the file with the bot handler class')
+
+    parser.add_argument('--force',
+                        action='store_true',
+                        help='Try running the bot even if dependencies install fails.')
+
+    parser.add_argument('--provision',
+                        action='store_true',
+                        help='Install dependencies for the bot.')
+
+    options = parser.parse_args()
+    if not options.name and not options.path_to_bot:
+        error_message = """
+You must either specify the name of an existing bot or
+specify a path to the file (--path-to-bot) that contains
+the bot handler class.
+"""
+        parser.error(error_message)
+
+    return options
+
+def main():
+    # type: () -> None
+    options = parse_args()
+    bot_name = options.name
+    if options.path_to_bot:
+        if options.provision:
+            bot_dir = os.path.dirname(os.path.abspath(options.path_to_bot))
+            provision_bot(bot_dir, options.force)
+        lib_module = import_module_from_source(options.path_to_bot, name=bot_name)
+    elif options.name:
+        if options.provision:
+            current_dir = os.path.dirname(os.path.abspath(__file__))
+            bots_parent_dir = os.path.join(current_dir, "bots")
+            bot_dir = os.path.join(bots_parent_dir, options.name)
+            provision_bot(bot_dir, options.force)
+        lib_module = import_module('zulip_bots.bots.{bot}.{bot}'.format(bot=bot_name))
+
+    message = {'content': options.message}
+    message_handler = lib_module.handler_class()
+
+    with patch('zulip_bots.lib.ExternalBotHandler') as mock_bot_handler:
+        def get_config_info(bot_name, section=None, optional=False):
+            # type: (str, Optional[str], Optional[bool]) -> Dict[str, Any]
+            conf_file_path = os.path.realpath(os.path.join(
+                'zulip_bots', 'bots', bot_name, bot_name + '.conf'))
```

Comment:
I think some bots use `bot_name.config`

Reply:
As a matter of fact, `converter` bot does not require a config file at all.
Will remove it.